instrumentation/github.com/labstack/echo/otelecho/config.go

```diff
@@ -4,17 +4,27 @@
 package otelecho // import "go.opentelemetry.io/contrib/instrumentation/github.com/labstack/echo/otelecho"
 
 import (
+	"github.com/labstack/echo/v4"
 	"github.com/labstack/echo/v4/middleware"
 
 	"go.opentelemetry.io/otel/propagation"
 	oteltrace "go.opentelemetry.io/otel/trace"
 )
 
+// defaultSpanNameFormatter is the default function used for formatting span names.
+var defaultSpanNameFormatter = func(c echo.Context) string {
+	return c.Path()
+}
+
+// SpanNameFormatter is a function that takes an echo.Context and returns a span name.
+type SpanNameFormatter func(c echo.Context) string
```

Comment:
I don't like how this takes in `echo.Context` and allows people to mutate it.

I would prefer `func(path string, r *http.Request) string` instead like in otelhttp https://github.com/open-telemetry/opentelemetry-go-contrib/blob/b508e33e7d8c2140cac4e9530ccd4e81b68be079/instrumentation/net/http/otelhttp/transport.go#L33

You would need to do `spanName := cfg.spanNameFormatter(c.Path(), c.Request())` instead. otherwise everything else looks great!